quick-start.sh

```diff
@@ -7,169 +7,347 @@
+spin() {
+  local -r title="${1}"
+  shift 1
+  gum spin -s line --title "${title}" -- $@
+}
 
-  echo "Please enter the authorization id of your Smarty Street Account."
-  echo "More info:   https://www.smarty.com/docs/cloud/authentication"
-  read SMARTY_AUTH_ID
+# Check if user has access to billing accounts and select one if they do
+enable_billing() {
+  BILLING_ACCOUNT_COUNT=$(gcloud beta billing accounts list --format json | jq '. | length')
+  while [ "$BILLING_ACCOUNT_COUNT" = "0" ]; do
+    echo "You don't have any $(pink 'billing accounts') yet."
+    echo "If you are responsible for billing, please create one in the Google Cloud Console at https://console.cloud.google.com/billing."
+    echo "If you are not responsible for billing, please ask your billing admin to create one for you."
+    echo "Press $(pink 'Enter') to continue once the billing account is created. Type $(pink 'exit') to exit the script."
+    read SHOULD_CONTINUE
+    if [ "$SHOULD_CONTINUE" = "exit" ]; then
+      exit 1
+    else
+      BILLING_ACCOUNT_COUNT=$(gcloud beta billing accounts list --format json | jq '. | length')
+    fi
+  done
 
-  echo "Please enter the authorization token of your Smarty Street Account."
-  echo "More info:   https://www.smarty.com/docs/cloud/authentication"
-  read SMARTY_AUTH_TOKEN
+  echo "You have $(pink $BILLING_ACCOUNT_COUNT) billing account(s)."
+  echo "Please select the $(pink 'billing account') you want to use for this project."
+  echo 
 
-  echo "Logging in to GitHub..."
-  gh auth login
+  BILLING_ACCOUNT_ID=$(gcloud beta billing accounts list --format="csv(displayName,name)" | gum table -w 25,25 | cut -d ',' -f 2)
+}
 
-  echo "Setting repository secrets..."
-  gh secret -R "${GITHUB_REPO}" set PROJECT_ID --body "${PROJECT_ID}" 
-  gh secret -R "${GITHUB_REPO}" set SERVICE_ACCOUNT_ID --body "${SERVICE_ACCOUNT_ID}"
-  gh secret -R "${GITHUB_REPO}" set WORKLOAD_IDENTITY_PROVIDER --body "${WORKLOAD_IDENTITY_PROVIDER}"
-  gh secret -R "${GITHUB_REPO}" set REGION --body "${REGION}"
-  gh secret -R "${GITHUB_REPO}" set ZONE --body "${ZONE}"
-  gh secret -R "${GITHUB_REPO}" set SMARTY_AUTH_ID --body "${SMARTY_AUTH_ID}"
-  gh secret -R "${GITHUB_REPO}" set SMARTY_AUTH_TOKEN --body "${SMARTY_AUTH_TOKEN}"
+# Link the selected billing account to the selected project
+link_billing_account() {
+  spin "Linking $(pink 'project') to billing account..." gcloud beta billing projects link "${PROJECT_ID}" --billing-account="${BILLING_ACCOUNT_ID}"
 
-  echo "Repository secrets set!"
-  echo "You can now continue with the Quick Start instructions in the README.md file."
 }
 
-# Prompt user for project name and repository name
-echo "Welcome to the PHDI Google Cloud setup script!"
-echo "This script will help you setup gcloud authentication for GitHub Actions."
+### Main ###
+
+# Install gum
+if ! command -v gum &> /dev/null; then
+    echo "Installing gum..."
+    go install github.com/charmbracelet/gum@latest
```

Comment:
Are we comfortable enough with the stability of GUM to use `latest` or should we target a specific release? I'm not entirely sure on this one, but lean slightly towards choosing a release on the off chance GUM ever introduces breaking changes. This would shift some maintenance burden onto our plate though. What are you thoughts?

Reply:
Hmmm, it seems like the kind of tool that's unlikely to introduce breaking changes, but there's no harm in targeting a specific version, either. I'll make that change!

Reply:
Could we do latest in the current minor version? Seems like a reasonable compromise if the unix magic for that isn't too hard.